tests/system/stats/stackdriver/stackdriver_stats_test.py

```diff
@@ -0,0 +1,121 @@
+                                                        client=client,
+                                                        transport=sync.SyncTransport)
+        view_manager.register_exporter(exporter)
+
+        # Register view.
+        view_manager.register_view(VIDEO_SIZE_VIEW)
+
+        # Sleep for [0, 10] milliseconds to fake work.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        # Process video.
+        # Record the processed video size.
+        tag_value = tag_value_module.TagValue("1200")
+        tag_map = tag_map_module.TagMap()
+        tag_map.insert(FRONTEND_KEY, tag_value)
+        measure_map = stats_recorder.new_measurement_map()
+        measure_map.measure_int_put(VIDEO_SIZE_MEASURE, 25 * MiB)
+
+        measure_map.record(tag_map)
+
+
+        # Sleep for [0, 10] milliseconds to fake wait.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        name = exporter.client.project_path(PROJECT)
+        list_metrics_descriptors = exporter.client.list_metric_descriptors(name)
+        element = [element for element in list_metrics_descriptors if element.description == "SampleViewDescriptionSync"]
+        self.assertTrue(any(element))
+
+    def test_stats_record_async(self):
+        FRONTEND_KEY_ASYNC = tag_key_module.TagKey("SampleKeyAsync")
+        VIDEO_SIZE_MEASURE_ASYNC = measure_module.MeasureInt(
+            "SampleMeasureNameAsync", "SampleDescriptionAsync", "By")
+        VIDEO_SIZE_VIEW_NAME_ASYNC = "SampleViewNameAsync"
+        VIDEO_SIZE_DISTRIBUTION_ASYNC = aggregation_module.DistributionAggregation(
+                                    [0.0, 16.0 * MiB, 256.0 * MiB])
+        VIDEO_SIZE_VIEW_ASYNC = view_module.View(VIDEO_SIZE_VIEW_NAME_ASYNC,
+                                        "SampleViewDescriptionAsync",
+                                        [FRONTEND_KEY_ASYNC],
+                                        VIDEO_SIZE_MEASURE_ASYNC,
+                                        VIDEO_SIZE_DISTRIBUTION_ASYNC)
+
+        stats = stats_module.Stats()
+        view_manager = stats.view_manager
+        stats_recorder = stats.stats_recorder
+
+        exporter = stackdriver.new_stats_exporter(stackdriver.Options(project_id=PROJECT))
+        view_manager.register_exporter(exporter)
+
+        # Register view.
+        view_manager.register_view(VIDEO_SIZE_VIEW_ASYNC)
+
+        # Sleep for [0, 10] milliseconds to fake work.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        # Process video.
+        # Record the processed video size.
+        tag_value = tag_value_module.TagValue("1200")
+        tag_map = tag_map_module.TagMap()
+        tag_map.insert(FRONTEND_KEY_ASYNC, tag_value)
+        measure_map = stats_recorder.new_measurement_map()
+        measure_map.measure_int_put(VIDEO_SIZE_MEASURE_ASYNC, 25 * MiB)
+
+        measure_map.record(tag_map)
+        self.assertTrue(True)
```

Comment:
This isn't test anything, maybe assert something meaningful here?

Reply:
In this case, there is an issue because we are testing in an async way. So, due to our async transport ( that wait for something like 5 seconds to send data ) we could not read it from Stackdriver api to verify if data was sent in a correct way.